src/rgw/rgw_aws.cc

```diff
@@ -48,16 +48,71 @@ namespace rgw::aws {
 
 
   };
+  class AsyncCallerContext: public Aws::Client::AsyncCallerContext{
+  public:
+    reply_callback_t cb;
+    CephContext* cct;
+    AsyncCallerContext(reply_callback_t cb, CephContext* cct):cb(cb), cct(cct){}
+  };
+
+  void invoke_callback(const Aws::Lambda::LambdaClient* client,
+                              const Aws::Lambda::Model::InvokeRequest& request,
+                              const Aws::Lambda::Model::InvokeOutcome & outcome,
+                              const std::shared_ptr<const Aws::Client::AsyncCallerContext>& context){
+
+    auto derived = std::static_pointer_cast<const AsyncCallerContext>(context);
+    CephContext* cct = derived->cct;
+    if (outcome.IsSuccess()) {
+      ldout(cct, 20) << "Invoke function: " << request.GetFunctionName() << " Response: " << outcome.GetResult().GetLogResult() << dendl;
+      if(derived && derived->cb) {
+        derived->cb(STATUS_OK);
+      }
+    }
+    else {
+      const auto& error = outcome.GetError();
+      ldout(cct, 20) << "ERROR: " << error.GetExceptionName() << ": "
+                << error.GetMessage() << dendl;
+      if(derived && derived->cb) {
+        derived->cb(-(int)error.GetErrorType());
```

Comment:
please avoid c-style casting. what is the return type of `GetErrorType()`?

Reply:
enum